### PR TITLE
Form buttons

### PR DESCRIPTION
**Pull request by ADmad**
### bugfix
- Set value for default "Save" button.
  Without a value you can't set a redirect reader (for RedirectListener) for it.

### enhancement
- Move default buttons config to separate method for easy extension.

### CODE REVIEW

File: src/Listener/Traits/FormTypeTrait.php [hunk 1]
```diff
@@ -56,33 +56,41 @@ protected function _getFormSubmitExtraButtons()
...
+        return [
+            'save_and_continue' => [
                 'title' => __d('crud', 'Save & continue editing'),
-                'options' => ['class' => 'btn btn-success btn-save-continue', 'name' => '_edit', 'value' => true],
+                'options' => ['class' => 'btn btn-success btn-save-continue', 'name' => '_edit', 'value' => '1'],
                 'type' => 'button',
                 '_label' => 'save_and_continue',
             ],
-            [
+            'save_and_create' => [
```
Comment (josegonzalez) [inline src/Listener/Traits/FormTypeTrait.php]: This seems like it might be backwards incompatible?

Comment (ADmad) [reply src/Listener/Traits/FormTypeTrait.php]: I highly doubt someone would have been relying on the numeric indexes. Also this list of defaults was moved from template to the this trait pretty recently so it's highly unlikely to cause BC problems.

Comment (josegonzalez) [reply src/Listener/Traits/FormTypeTrait.php]: Would it break the existing form element?

Comment (ADmad) [reply src/Listener/Traits/FormTypeTrait.php]: Nope it won't. This array is only used in `Element/form/buttons.ctp` and this change won't cause any problem.

**Pull request merged**